src/main/java/com/auth0/utils/tokens/IdTokenVerifier.java

```diff
@@ -134,6 +136,15 @@ public void verify(String token, String nonce, Integer maxAuthenticationAge) thr
             throw new IdTokenValidationException(String.format("Audience (aud) claim mismatch in the ID token; expected \"%s\" but found \"%s\"", this.audience, decoded.getAudience()));
         }
 
+        // Org verification
+        if (this.organization != null) {
+            String orgClaim = decoded.getClaim("org_id").asString();
+            if (!this.organization.equals(orgClaim)) {
+                throw new IdTokenValidationException(String.format("Organization (org_id) claim mismatch in the ID token; expected \"%s\" but found \"%s\"", this.organization, orgClaim));
+            }
+
+        }
```

Comment:
Should we have a comparable validation as we have with Audience?

- check if it exists and is a string
- Check if it matched ?

```suggestion
        if (this.organization != null) {
            String orgClaim = decoded.getClaim("org_id").asString();
            if (isEmpty(orgClaim)) {
               throw new IdTokenValidationException("Organization (org_id) claim must be a string present in the ID token");
            }
            if (!this.organization.equals(orgClaim)) {
                throw new IdTokenValidationException(String.format("Organization (org_id) claim mismatch in the ID token; expected \"%s\" but found \"%s\"", this.organization, orgClaim));
            }
        }
```

Edit: suggestion diff is a bit odd here ... I added 
```
if (isEmpty(orgClaim)) {
    throw new IdTokenValidationException("Organization (org_id) claim must be a string present in the ID token");
}
```

Reply:
Yes, this makes sense. I'll push a new commit with these changes.

Reply:
Added check for null/empty in new commit.